### ISSUE TITLE
Async Support

### ISSUE THREAD

**Issue by ido-ran** [ligi/ipfs-api-kotlin#5]
Add commands are currently executing synchronously which means it is not possible to call them from the UI thread.
Do you think it makes sense to add async support or since any operation will take over 100ms there is no point in initiating them on the UI thread?

**Comment by ligi** [comment]
Yes - but I would not bake it into the core api - but offer it as modules to not become too opinionated and bloated. Modules could be called coroutines or rx
